src/acquisition/covidcast/test_utils.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+
+from delphi_utils import Nans
+from delphi.epidata.acquisition.covidcast.database import Database, CovidcastRow
+import delphi.operations.secrets as secrets
+
+# all the Nans we use here are just one value, so this is a shortcut to it:
+nmv = Nans.NOT_MISSING.value
+
+class CovidcastBase(unittest.TestCase):
+    def setUp(self):
+        # use the local test instance of the database
+        secrets.db.host = 'delphi_database_epidata'
+        secrets.db.epi = ('user', 'pass')
+
+        self._db = Database()
+        self._db.connect()
+
+        # empty all of the data tables
+        for table in "signal_load  signal_latest  signal_history  geo_dim  signal_dim".split():
+            self._db._cursor.execute(f"TRUNCATE TABLE {table};")
+        self.localSetUp()
+        self._db._connection.commit()
+
+    def localSetUp(self):
+        pass
```

Comment:
Todo or local needs no setup? If the latter, why is it called in setup?

Also, in the case that a test fails or a test doesn't clean up after itself, it seems like it may be beneficial for the truncate to happen on the local DB too.

Reply:
it's a stub, available for overriding in subclasses.

Reply:
```suggestion
        # stub; override in subclasses to perform custom setup. 
        # runs after tables have been truncated but before database changes have been committed
        pass
```

Reply:
this method is a placeholder, and subclasses of this base class can redefine/override it if they choose.  that way, those subclasses can specify "local" set-up procedures that are applicable only to themselves and would be meaningless (or even inappropriate) for a different test class.  its sort of like an abstract method, but optional.

any expectation for "cleaning up" the state of the tables is really covered in `setUp()`: it runs before each test case and provides a clean slate for that test.  anything that test leaves behind (populated rows) will be removed for the next test when `setUp()` runs again.

Reply:
@krivard im fine with adding that suggested change, especially because its small and only documentation, but no more [riders](https://en.wikipedia.org/wiki/Rider_(legislation))!